array of interfaces

### The following program `sample.go` triggers an unexpected result

```go
package main

import "io"

type B []byte

func (b B) Write(p []byte) (n int, err error) {
    b = p 
    return len(p), nil 
}

func main() {
    b := B{} 
    a := make([]io.Writer, 0)
    a = append(a, b)
    println(len(a))
}
```


### Expected result

```console
$ go run ./sample.go
1
```


### Got

```console
$ yaegi ./sample.go
/./sample.go:17:6: panic
run: reflect.Set: value of type []uint8 is not assignable to type io.Writer
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0x140002e6a00, 0x40)
	/Users/marc/sdk/go1.16.6/src/runtime/debug/stack.go:24 +0x88
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0x140001afbf0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/interp.go:504 +0x94
panic(0x102c09f20, 0x1400019b200)
	/Users/marc/sdk/go1.16.6/src/runtime/panic.go:965 +0x14c
github.com/traefik/yaegi/interp.runCfg.func1(0x140000cabb0, 0x140004b5200, 0x140001af9c0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:185 +0x1e8
panic(0x102c09f20, 0x1400019b200)
	/Users/marc/sdk/go1.16.6/src/runtime/panic.go:965 +0x14c
reflect.Value.assignTo(0x102c00f60, 0x1400000ed08, 0x197, 0x102a2789f, 0xb, 0x102c5f9a0, 0x1400019b1f0, 0x97, 0x1400000edc8, 0x102bfeb20)
	/Users/marc/sdk/go1.16.6/src/reflect/value.go:2451 +0x2d0
reflect.Value.Set(0x102c5f9a0, 0x1400019b1f0, 0x194, 0x102c00f60, 0x1400000ed08, 0x197)
	/Users/marc/sdk/go1.16.6/src/reflect/value.go:1564 +0x94
reflect.Append(0x102bfeb20, 0x1400000edb0, 0x197, 0x140001af970, 0x1, 0x1, 0x0, 0x102bfeb20, 0x1400000ed20)
	/Users/marc/sdk/go1.16.6/src/reflect/value.go:2051 +0xac
github.com/traefik/yaegi/interp._append.func2(0x140000cabb0, 0x140003e1830)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:3020 +0xf0
github.com/traefik/yaegi/interp.runCfg(0x140004b5200, 0x140000cabb0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:191 +0x8c
github.com/traefik/yaegi/interp.(*Interpreter).run(0x14000368000, 0x140004b4a00, 0x140000caa50)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/run.go:122 +0x244
github.com/traefik/yaegi/interp.(*Interpreter).eval(0x14000368000, 0x140004b20f0, 0xed, 0x16dc277e8, 0x21, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/interp.go:596 +0x5e8
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0x14000368000, 0x16dc277e8, 0x21, 0x140004b2000, 0xed, 0x0, 0x0, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi/interp/interp.go:418 +0xc4
main.runFile(0x14000368000, 0x16dc277e8, 0x21, 0x0, 0x0, 0x0)
	/Users/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/run.go:149 +0x84
main.run(0x14000032050, 0x1, 0x1, 0x14000001380, 0x1400006eea8)
	/Users/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/run.go:112 +0x830
main.main()
	/Users/marc/go/src/github.com/traefik/yaegi/cmd/yaegi/yaegi.go:144 +0x324
```


### Yaegi Version

v0.9.19

### Additional Notes

_No response_

--- a/_test/issue-1185.go
+++ b/_test/issue-1185.go
@@ -1,0 +1,20 @@
+package main
+
+import "io"
+
+type B []byte
+
+func (b B) Write(p []byte) (n int, err error) {
+	b = p
+	return len(p), nil
+}
+
+func main() {
+	b := B{}
+	a := make([]io.Writer, 0)
+	a = append(a, b)
+	println(len(a))
+}
+
+// Output:
+// 1

--- a/interp/run.go
+++ b/interp/run.go
@@ -2984,6 +2984,8 @@ func _append(n *node) {
 				values[i] = genValue(arg)
 			case isInterfaceSrc(n.typ.val):
 				values[i] = genValueInterface(arg)
+			case isInterfaceBin(n.typ.val):
+				values[i] = genInterfaceWrapper(arg, n.typ.val.rtype)
 			case isRecursiveType(n.typ.val, n.typ.val.rtype):
 				values[i] = genValueRecursiveInterface(arg, n.typ.val.rtype)
 			case arg.typ.untyped:
@@ -3008,6 +3010,8 @@ func _append(n *node) {
 			value0 = genValue(n.child[2])
 		case isInterfaceSrc(elem):
 			value0 = genValueInterface(n.child[2])
+		case isInterfaceBin(elem):
+			value0 = genInterfaceWrapper(n.child[2], elem.rtype)
 		case isRecursiveType(elem, elem.rtype):
 			value0 = genValueRecursiveInterface(n.child[2], elem.rtype)
 		case n.child[2].typ.untyped:

